File: backend/modules/evaluation/domain/entity/evaluator.go
// Copyright (c) 2025 Bytedance Ltd. and/or its affiliates
// SPDX-License-Identifier: Apache-2.0

package entity

type Evaluator struct {
	ID             int64
	SpaceID        int64
	Name           string
	Description    string
	DraftSubmitted bool
	EvaluatorType  EvaluatorType
	LatestVersion  string
	BaseInfo       *BaseInfo

	PromptEvaluatorVersion *PromptEvaluatorVersion
}

type EvaluatorType int64

const (
	EvaluatorTypePrompt EvaluatorType = 1
	EvaluatorTypeCode   EvaluatorType = 2
)

var EvaluatorTypeSet = map[EvaluatorType]struct{}{
	EvaluatorTypePrompt: {},
	EvaluatorTypeCode:   {},
}

func (e *Evaluator) GetEvaluatorVersion() IEvaluatorVersion {
	switch e.EvaluatorType {
	case EvaluatorTypePrompt:
		return e.PromptEvaluatorVersion
	default:
		return nil
	}
}

func (e *Evaluator) SetEvaluatorVersion(version *Evaluator) {
	switch e.EvaluatorType {
	case EvaluatorTypePrompt:
		e.PromptEvaluatorVersion = version.PromptEvaluatorVersion
	default:
		return
	}
}

File: frontend/packages/cozeloop/evaluate/src/request/evaluator.ts
// Copyright (c) 2025 Bytedance Ltd. and/or its affiliates
// SPDX-License-Identifier: Apache-2.0
import {
  type GetEvaluatorVersionRequest,
  type GetEvaluatorVersionResponse,
} from '@cozeloop/api-schema/evaluation';
import { StoneEvaluationApi } from '@cozeloop/api-schema';

export async function getEvaluatorVersion(
  params: GetEvaluatorVersionRequest,
): Promise<GetEvaluatorVersionResponse> {
  return StoneEvaluationApi.GetEvaluatorVersion(params);
}
